### PR TITLE
NFV SRIOV document changes

### PR DESCRIPTION
**Pull request by Jaganathancse**
This change is to add the deployment guide for NFV SRIOV.

### CODE REVIEW

File: nfv_sriov.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+kind: OpenStackDataPlaneService
+metadata:
+  name: nova-custom-sriov
+spec:
+  label: dataplane-deployment-nova-custom-sriov
+  configMaps:
+    - cpu-pinning-nova
+    - sriov-nova
+  secrets:
+    - nova-cell1-compute-config
```
Comment (abays) [inline nfv_sriov.md:160]: These have a new format [1] for inclusion in `OpenStackDataPlaneService`.

[1] https://github.com/slagle/architecture/blob/ffbbba2cbd989dd18489c5bd43f83e0f344737c7/va/nfv/sriov/edpm/nodeset/nova_sriov.yaml#L22-L30

File: nfv_sriov.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+The `nova-custom-sriov` can be seen in the
+[example](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+and takes the place of the default `nova`
+OpenStackDataPlaneService. This custom service uses a `ConfigMap` called
+`cpu-pinning-nova` which ensures that the file `03-cpu-pinning-nova.conf` is used
+by Nova.
+
+Now that the `nova-custom-sriov` has been created, use the example
+[dataplane_v1beta1_openstackdataplanedeployment_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+to start the second deployment.
```
Comment (abays) [inline nfv_sriov.md:246]: What is the second deployment?  Isn't there only one `OpenStackDataPlaneDeployment`?

Comment (jpodivin) [reply nfv_sriov.md:246]: Nothing is preventing us to start multiple deployments.

File: nfv_sriov.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+`OpenStackDataPlaneNodeSet` CR needs to be edited.
+
+## OpenStackDataPlaneNodeSet services list
+
+EDPM nodes can be configured by creating an
+`OpenStackDataPlaneNodeSet` CR which the
+[dataplane-operator](https://openstack-k8s-operators.github.io/dataplane-operator)
+will reconcile to create OpenStackDataPlaneService resources
+when an `OpenStackDataPlaneDeployment` CR is created.
+These types of CRs have a `services` list like the following:
```
Comment (jpodivin) [inline nfv_sriov.md:22]: It is apparent from the example. But simply `OpenStackDataPlaneNodeSet` would work here.

File: nfv_sriov.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    - reboot-os
+    - install-certs
+    - libvirt
+    - ovn
+    - neutron-ovn
+    - nova-custom-sriov
+    - neutron-sriov
+    - neutron-metadata
+```
+Only the services which are on the list will be configured.
```
Comment (jpodivin) [inline nfv_sriov.md:47]: Please add a note about `OpenStackDataPlaneService` CRs, explaining that these are the services being created. We must avoid confusion with other "services" due to the overloaded terminology.

File: nfv_sriov.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+Following are the list of EDPM ansible variables which need to be
+provided for deploying with SRIOV support.
+```
+edpm_kernel_args: Configures kernel boot parameters
+
+edpm_tuned_profile: tuned profile configuration
+
+edpm_tuned_isolated_cores: List of isolated cores
+
+edpm_nova_libvirt_qemu_group: Configures qemu group for nova libvirt
```
Comment (jpodivin) [inline nfv_sriov.md:60]: Please link to the underlying role in edpm-ansible. We need people to have easy way to reach the source where more information is provided.

File: nfv_sriov.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+edpm_network_config_template
+  -
+    type: sriov_pf
+    name: nic3
+    numvfs: 10
+    use_dhcp: false
+    promisc: true
+```
+
+This example also assumes that the EDPM nodes:
```
Comment (jpodivin) [inline nfv_sriov.md:84]: Please turn it into note and move closer to the headline. Ideally between list of vars and example itself.

File: nfv_sriov.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+- PXE and boot settings are already configured.
+- Are at least one baremetal compute with SRIOV interfaces.
+
+Create an `OpenStackDataPlaneNodeSet` CR file,
+e.g. `dataplane_cr.yaml` to represent the EDPM nodes. See
+[dataplane_v1beta1_openstackdataplanenodeset.yaml](https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml)
+for an example to modify as described in this document.
+
+Do not yet create the CR in OpenShift as the edits described in the
+next sections are required.
```
Comment (jpodivin) [inline nfv_sriov.md:95]: Please highlight the warning. People may miss it.

File: nfv_sriov.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+### Create a OpenStackDataPlaneDeployment
+
+Creating an `OpenStackDataPlaneDeployment` will trigger Ansible jobs
+to configure an EDPM. Which Ansible roles are run depends on the
+`services` list.
+
+Each `OpenStackDataPlaneDeployment` can have its own
+`servicesOverride` list which will redefine the list
+of services of an `OpenStackDataPlaneNodeSet` for a
+deployment.
```
Comment (jpodivin) [inline nfv_sriov.md:224]: Not just of one nodeset. There can be more than one. In fact we expect that to be the case relatively often.

**Review by jpodivin — CHANGES_REQUESTED** [review]
Several suggestions, I would also like to ask @vakwetu to take a look, as there were recent changes to certs.

File: nfv_sriov.md (hunk 2 above)
Comment (abays) [reply nfv_sriov.md:246]: Yes, true.  But where is the need for a second deployment in the process being described here?

File: nfv_sriov.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+
+The example
+[dataplane_v1beta1_openstackdataplanedeployment_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+
+Create the CR based on the example
+```
+oc kustomize --load-restrictor LoadRestrictionsNone dataplane-operator/examples/sriov > dataplane_cr.yaml
+```
+
+Custom `OpenStackDataPlaneService` called `nova-custom-sriov`
```
Comment (vakwetu) [inline nfv_sriov.md:234]: I'm a little confused here.  I followed the links but its not clear to me what the nova-custom-sriov dataplane service definition should look like.

Comment (vakwetu) [reply nfv_sriov.md:234]: never mind .. need to look up ..

File: nfv_sriov.md (hunk 1 above)
Comment (vakwetu) [reply nfv_sriov.md:160]: On that note, there are several fields in that definition that relate to certs.  I'm not sure if you actually need the things under tlsCert -- I suspect not - but if you do, then that format has recently changed.  
https://github.com/openstack-k8s-operators/dataplane-operator/pull/875

On the other hand, you probably also need the caCerts field to mount and trust the caCerts so you can talk to other services.

Comment (Jaganathancse) [reply nfv_sriov.md]: done

File: nfv_sriov.md (hunk 2 above)
Comment (Jaganathancse) [reply nfv_sriov.md]: only one deployment. removed it.

File: nfv_sriov.md (hunk 3 above)
Comment (Jaganathancse) [reply nfv_sriov.md:22]: done

File: nfv_sriov.md (hunk 4 above)
Comment (Jaganathancse) [reply nfv_sriov.md:47]: done

File: nfv_sriov.md (hunk 5 above)
Comment (Jaganathancse) [reply nfv_sriov.md:62]: only added mandatory parameters. i think no need to specify edpm-ansible role parameters here.

File: nfv_sriov.md (hunk 6 above)
Comment (Jaganathancse) [reply nfv_sriov.md]: done

File: nfv_sriov.md (hunk 7 above)
Comment (Jaganathancse) [reply nfv_sriov.md:96]: done

File: nfv_sriov.md (hunk 8 above)
Comment (Jaganathancse) [reply nfv_sriov.md]: done

**Review by abays — APPROVED** [review]
/lgtm

**Review by jpodivin — APPROVED** [review]

**Pull request merged**